spec/features/user_edits_profile_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+feature 'user edits profile' do
+  let!(:user) { FactoryGirl.create(:user) }
+  before(:each) { login_as(user, scope: :user) }
+  after(:each) { logout(:user) }
+
+  scenario 'user goes to edit profile page' do
+    visit '/'
+    click_on 'Edit profile'
+    expect(page).to have_content('Edit User')
+  end
+
+  scenario 'form has correct fields' do
+    visit edit_user_registration_path
+
+    expect(page).to have_content 'Username'
+    expect(page).to have_content 'Email'
+    expect(page).to have_content 'Password'
+    expect(page).to have_content 'Confirm Password'
+    expect(page).to have_content 'Current Password'
+    expect(page).to have_button('Update')
+  end
+
+  scenario 'user fills out form with valid values and submits' do
+    visit edit_user_registration_path
+    id = user.id
+    fill_in 'Username', with: 'rustyshackleford'
+    fill_in 'Email', with: 'dale@gribble.com'
+    fill_in 'Password', with: 'testtest'
+    fill_in 'Confirm Password', with: 'testtest'
+    fill_in 'Current Password', with: user.password
+
+    click_on 'Update'
+
+    user = User.find(id)
+    expect(page).to have_content 'Your account has been updated successfully.'
+    expect(user.username).to eq 'rustyshackleford'
+    expect(user.email).to eq 'dale@gribble.com'
+  end
+
+  scenario 'user fills out form with invalid values' do
+    visit edit_user_registration_path
+    id = user.id
+    fill_in 'Username', with: 'a'
+    fill_in 'Email', with: 'dalegribble.com'
+    fill_in 'Password', with: 'testtest'
+    fill_in 'Confirm Password', with: 'testtest'
+    fill_in 'Current Password', with: user.password
```

Comment:
Best practice here, since you just have one general error message that you're rendering on your page, would be to make a separate test for each failing field -- the way this is tested currently, you don't know for sure that each (username and email) is failing, just that ONE of them is failing. Can you separate out into two separate tests, one for each field here?